src/main/webapp/script.js

```diff
@@ -30,43 +27,134 @@ function passQuery() {
+  am4core.useTheme(am4themes_animated);
+  const chart = am4core.create('map', am4maps.MapChart);
+  chart.height = 550;
+  chart.zoomControl = new am4maps.ZoomControl();
+  // only allow zooming with buttons
+  chart.mouseWheelBehavior = 'none';
+
+  // Create map instance
+  if (isCountyQuery) {
+    chart.geodata = location ===
+        '06' ? am4geodata_region_usa_caLow : am4geodata_region_usa_njLow;
+  } else {
+    chart.geodata = am4geodata_usaLow;
+  }
+
+  // Add button to zoom out
+  const home = chart.chartContainer.createChild(am4core.Button);
+  home.label.text = 'Home';
+  home.align = 'right';
+  home.events.on('hit', function(ev) {
+    chart.goHome();
+  });
+
+  chart.projection = new am4maps.projections.AlbersUsa();
+  const polygonSeries = chart.series.push(new am4maps.MapPolygonSeries());
+
+  // Set min/max fill color for each area
+  polygonSeries.heatRules.push({
+    property: 'fill',
+    target: polygonSeries.mapPolygons.template,
+    min: chart.colors.getIndex(1).brighten(1),
+    max: chart.colors.getIndex(1).brighten(-0.3),
+    logarithmic: true,
+  });
+  polygonSeries.useGeodata = true;
+  polygonSeries.data = data;
+
+  // Set up heat legend
+  const heatLegend = chart.createChild(am4maps.HeatLegend);
+  heatLegend.series = polygonSeries;
+  heatLegend.align = 'right';
+  heatLegend.valign = 'bottom';
+  heatLegend.height = am4core.percent(80);
+  heatLegend.orientation = 'vertical';
+  heatLegend.valign = 'middle';
+  heatLegend.marginRight = am4core.percent(4);
+  heatLegend.valueAxis.renderer.opposite = true;
+  heatLegend.valueAxis.renderer.dx = - 25;
+  heatLegend.valueAxis.strictMinMax = false;
+  heatLegend.valueAxis.fontSize = 9;
+  heatLegend.valueAxis.logarithmic = true;
+
+  // Configure series tooltip
+  const polygonTemplate = polygonSeries.mapPolygons.template;
+  let descriptionString = 'Population of people ' +
+    description.age + ' who ' + description.action;
+  // make it moved to
+  if (description.action === 'moved') {
+    descriptionString += ' to';
+  }
+  polygonTemplate.tooltipText = '{name}\n' + descriptionString + ': {value}';
+  polygonTemplate.nonScalingStroke = true;
+  polygonTemplate.strokeWidth = 0.5;
+
+  // Create hover state and set alternative fill color
+  const hs = polygonTemplate.states.create('hover');
+  hs.properties.fill = am4core.color('#3c5bdc');
+
+
+  // heat legend behavior
+  polygonSeries.mapPolygons.template.events.on('over', function(event) {
+    handleHover(event.target);
+  });
+  polygonTemplate.events.on('hit', function(event) {
+    chart.zoomToMapObject(event.target);
+  });
+  function handleHover(column) {
+    if (!isNaN(column.dataItem.value)) {
+      heatLegend.valueAxis.showTooltipAt(column.dataItem.value);
+    } else {
+      heatLegend.valueAxis.hideTooltip();
+    }
+  }
+  polygonSeries.mapPolygons.template.events.on('out', function(event) {
+    heatLegend.valueAxis.hideTooltip();
+  });
+}
+
 function validateInput(dataListId) {
```

Comment:
I think you might have accidentally deleted this function's description (on the old lines 68 and 69).

Reply:
Thanks for catching this!